interface/ceed-basis.c

```diff
@@ -1107,85 +1152,60 @@ int CeedBasisGetInterp(CeedBasis basis, CeedScalar **interp) {
   @ref Advanced
 **/
 int CeedBasisGetGrad(CeedBasis basis, CeedScalar **grad) {
-  *grad = basis->grad1d;
+  if (!basis->grad && basis->tensorbasis) {
+    // Allocate
+    int ierr;
+    ierr = CeedMalloc(basis->dim*basis->Q*basis->P, &basis->grad);
+    CeedChk(ierr);
+
+    // Initalize
+    for (CeedInt i=0; i<basis->dim*basis->Q*basis->P; i++)
+      basis->grad[i] = 1.0;
+
+    // Calculate
+    for (CeedInt d=0; d<basis->dim; d++) {
+      for (CeedInt i=0; i<basis->dim; i++) {
+        CeedPragmaSIMD
+        for (CeedInt qpt=0; qpt<basis->Q; qpt++) {
+          CeedPragmaSIMD
+          for (CeedInt node=0; node<basis->P; node++) {
```

Comment:
I think these pragmas are also problematic.

If you're going to have the naive code below rather than hoisting it out/restructuring the loop, it's probably best to skip all pragmas because they imply that this code is meant to be high-performance.